astropy/cosmology/core.py

```diff
@@ -1388,7 +1388,7 @@ def angular_diameter_distance_z1z2(self, z1, z2):
           pair.
 
         """
-
+        if (z2 <= z1) : raise ValueError("Second redshift z2 must be larger than first redshift z1")
```

Comment:
Just nitpick to make it slightly more legible and informative.
```suggestion
        if (z2 <= z1):
            raise ValueError(f"Second redshift z2 ({z2}) must be larger than first redshift z1 ({z1})")
```